apps/dotcom/sync-worker/src/UserDataSyncer.ts

```diff
@@ -196,84 +196,35 @@ export class UserDataSyncer {
 		this.broadcast({ type: 'commit', mutationIds: mutationIds })
 	}
 
-	private updateStateAfterBootStep() {
-		assert(this.state.type === 'connecting', 'state should be connecting')
-		if (this.state.data) {
-			if (this.state.didGetBootId) {
-				// we got everything, so we can set the state to connected and apply any buffered events
-				const promise = this.state.promise
-				const bufferedEvents = this.state.bufferedEvents
-				const data = this.state.data
-				const mutationNumber = this.state.mutationNumber
-				this.state = {
-					type: 'connected',
-					bootId: this.state.bootId,
-					sequenceId: this.state.sequenceId,
-					lastSequenceNumber: this.state.lastSequenceNumber,
-				}
-				this.store.initialize(data)
-
-				if (bufferedEvents.length > 0) {
-					bufferedEvents.forEach((event) => this.handleReplicationEvent(event))
-					this.broadcast({
-						type: 'initial_data',
-						initialData: assertExists(this.store.getCommittedData()),
-					})
-				}
-				promise.resolve(null)
-
-				this.commitMutations(mutationNumber)
-			} else {
-				this.broadcast({
-					type: 'initial_data',
-					initialData: this.state.data,
-				})
-			}
+	private async loadInitialDataFromR2() {
+		this.log.debug('loading snapshot from R2')
+		const res = await this.env.UPLOADS.get(this.getSnapshotKey())
```

Comment:
I'd feel more comfortable if we didn't reuse the uploads bucket. We have some api routes that allow fetching stuff from the bucket, so I'm afraid we might expose these snapshots down the line.

(for now it seems fine, using an url like this doesn't seem to work, but only because our router does not match the path https://pr-5497-preview-deploy.tldraw.com/api/app/uploads/user_do_snapshots/user_id).

Reply:
yeah I meant to make a new bucket before submitting for review but it slipped my mind. will do that now.